proxy/http/HttpTransact.cc

```diff
@@ -7586,6 +7596,12 @@ HttpTransact::build_request(State *s, HTTPHdr *base_request, HTTPHdr *outgoing_r
     TxnDebug("http_trans", "[build_request] request expect 100-continue headers removed");
   }
 
+#if TS_HAS_TLS_EARLY_DATA
```

Comment:
I'm not too sure if we should surround this block.
It'd make more sense if `is_early_data` doesn't exist, but currently it exists regardless of `TS_HAS_TLS_EARLY_DATA`. I'd rather make sure that `is_early_data()` returns `false` if `TS_HAS_TLS_EARLY_DATA` is false to have less `#if` preprocessor directives, which makes code hard to read.

Reply:
fixed